pkg/operator/apiserver/controllerset/apiservercontrollerset.go

```diff
@@ -276,6 +288,62 @@ func (cs *APIServerControllerSet) WithoutRevisionController() *APIServerControll
 	return cs
 }
 
+func (cs *APIServerControllerSet) WithEncryptionControllers(
+	component string,
+	encryptedGRs []schema.GroupResource,
+	migrationClient kubemigratorclient.Interface,
+	kubeClient kubernetes.Interface,
+	apiServerClient configv1client.APIServerInterface,
+	secretsClient corev1client.SecretsGetter,
+	apiServerInformer configv1informers.APIServerInformer,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	resourceSyncController resourcesynccontroller.ResourceSyncer,
+	deployer statemachine.Deployer,
+) *APIServerControllerSet {
+
+	if deployer == nil {
+		var err error
+		nodeInformer := kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes()
+		nodeProvider := apiserveroperator.DeploymentNodeProvider{
+			TargetNamespace:                   component,
+			TargetNamespaceDeploymentInformer: kubeInformersForNamespaces.InformersFor(component).Apps().V1().Deployments(),
+			NodeInformer:                      nodeInformer,
+		}
+
+		deployer, err = encryptiondeployer.NewRevisionLabelPodDeployer("revision", component, kubeInformersForNamespaces, resourceSyncController, kubeClient.CoreV1(), kubeClient.CoreV1(), nodeProvider)
+		if err != nil {
+			err = fmt.Errorf("unable to create encryption controllers because failed to create a deployer, due to %v", err)
+			cs.unhandledErrors = append(cs.unhandledErrors, err)
+			return cs
+		}
+	}
+
+	migrationInformer := migrationv1alpha1informer.NewSharedInformerFactory(migrationClient, time.Minute*30)
```

Comment:
why create the informer here instead of it being passed as an argument (and thus not having to special-case it for informer start)?

Reply:
if we do that, then clients would have to deal with "migrator" - wanted to avoid that as they don't need to know about it. Does it make sense?

Reply:
They need to pass the migration client anyway, I'd say pass it as an argument in the name of uniform informer handling

Reply:
okay, but I will move `deployer` as well, so that dependencies are created in the main program.